docs/maintain-guides-how-to-upgrade.md

```diff
@@ -0,0 +1,51 @@
+---
+id: maintain-guides-how-to-upgrade
+title: How to Upgrade Your Validator
+sidebar_label: How to Upgrade Your Validator
+---
+
+Validators perform critical functions for the network, and as such, have strict uptime requirements. Validators may have to go offline for periods of time to upgrade the client software or the host machine. This guide will walk you through upgrading your machine and keeping your validator online.
+
+The process will take several hours, so make sure you understand the instructions first and plan accordingly.
+
+## Key Components
+
+### Session Keys
+
+Session keys are stored in the client and used to sign validator operations. They are what link your validator node to your Controller account. You cannot change them mid-Session.
+
+[More info about keys in Polkadot.](learn-keys)
+
+### Database
+
+Validators keep a database with all of their votes. If you start another machine with a validator on it, it is important to sync the database with your other machine or you risk equivocating.
+
+[More info about equivocation.](learn-staking#slashing)
+
+## Steps
+
+You will need to start a second validator to operate while you upgrade your primary. Throughout these steps, we will refer to the validator that you are upgrading as "Validator A" and the second one as "Validator B."
+
+### Session `N`
+
+1. Start a second node and connect it to your sentry nodes. Ideally, you use the same database as Validator A. Once it is synced, use the `--validator` flag. This is "Validator B."
+1. Generate Session keys in Validator B.
+1. Submit a `set_key` extrinsic from your Controller account with your new Session keys.
+1. Take note of the Session that this extrinsic was executed in.
+
+It is imperitive that your Validator A keep running in this Session. `set_key` only takes effect in the next Session.
+
+### Session `N+1`
+
+Validator B is now acting as your validator. You can safely take Validator A offline.
+
+1. Stop Validator A.
```

Comment:
Maybe add a note here about making sure that any block in the current session `N + 1` has been finalized. Otherwise `Validator A` might still be active in the GRANDPA authority set (since it hasn't finalized the end session block and thus hasn't transitioned).

A message like:

```
2019-10-28 21:44:13 Applying authority set change scheduled at block #450092
2019-10-28 21:44:13 Applying GRANDPA set change to new set with 20 authorities
```

Should also show up in the node logs.